torch/distributed/_tensor/_dispatch.py

```diff
@@ -113,9 +115,11 @@ def dispatch(
 
         # extract local tensor and sharding infos to a OpInfo
         op_info = self.unwrap_to_op_info(op_call, args, kwargs)
+        logger.debug("Dispatching op_call: %s", op_info.schema)
 
         self.sharding_propagator.propagate(op_info)
         output_sharding = op_info.output_sharding
+        logger.debug("output_sharding for %s: %s", op_call, output_sharding)
```

Comment:
Curious: is it possible to send `output` sharding to a `CommDebugMode` object if a `CommDebugMode` `TorchDispatchMode` is present?

Reply:
I think it is possible. CommDebugMode can hijack the logger.debug and get the correct output_sharding.